### PR TITLE
Added filtered disk index readme

### PR DESCRIPTION
**Pull request by NeelamMahapatro**
- Added readme for filtered disk index build
- default universal label 0 for memory index
- Removed duplicate assertion statement

- [ ] Does this PR have a descriptive title that could go in our release notes?
- [ ] Does this PR add any new dependencies?
- [ ] Does this PR modify any existing APIs?
   - [ ] Is the change to the API backwards compatible?
- [ ] Should this result in any changes to our documentation, either updating existing docs or adding new ones?

### CODE REVIEW

File: workflows/filtered_in_ssd.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+9. **-K**: search for *K* neighbors and measure *K*-recall@*K*, meaning the intersection between the retrieved top-*K* nearest neighbors and ground truth *K* nearest neighbors.
+10. **--result_path**: Search results will be stored in files with specified prefix, in bin format.
+11. **-L (--search_list)**: A list of search_list sizes to perform search with. Larger parameters will result in slower latencies, but higher accuracies. Must be atleast the value of *K* in arg (9).
+12. **--filter_label**: The filter to be used when searching an index with filters. For each query, a search is performed with this filter.
```
Comment (dkaczynski) [inline workflows/filtered_in_ssd.md:57]: Can you add a begin-to-end command line sample, like this? https://github.com/microsoft/DiskANN/blob/main/workflows/filtered_in_memory.md?plain=1#L66

I think we can use the same sample, except with `build_disk_index` and `search_disk_index` and its args

**Review by dkaczynski — APPROVED** [review]
At some point, we'll want to refine the synthetic label generation to demonstrate better recall rates, but that's not an issue in scope of this PR.

Approved.

File: README.md [hunk 2]
```diff
@@ -87,6 +87,7 @@ Please see the following pages on using the compiled code:
 - [Commandline interface for building and search in memory indices](workflows/in_memory_index.md) 
 - [Commandline examples for using in-memory streaming indices](workflows/dynamic_index.md)
 - [Commandline interface for building and search in memory indices with label data and filters](workflows/filtered_in_memory.md)
+- [Commandline interface for building and search SSD based indices with label data and filters](workflows/filtered_in_ssd.md)
```
Comment (harsha-simhadri) [inline README.md:90]: please rename filtered_ssd_index.md

File: src/index.cpp [hunk 3]
```diff
@@ -561,7 +561,6 @@ void Index<T, TagT, LabelT>::load(const char *filename, uint32_t num_threads, ui
         if (file_exists(labels_to_medoids))
         {
             std::ifstream medoid_stream(labels_to_medoids);
-            assert(label_num_pts == data_file_num_pts);
```
Comment (harsha-simhadri) [inline src/index.cpp:564]: Why is this assert removed?

File: tests/build_memory_index.cpp [hunk 4]
```diff
@@ -52,7 +52,7 @@ int build_in_memory_index(const diskann::Metric &metric, const std::string &data
         convert_labels_string_to_int(label_file, labels_file_to_use, mem_labels_int_map_file, universal_label);
         if (universal_label != "")
         {
-            LabelT unv_label_as_num = std::stoul(universal_label);
+            LabelT unv_label_as_num = 0;
```
Comment (harsha-simhadri) [inline tests/build_memory_index.cpp:55]: What is the change for?

File: workflows/filtered_in_ssd.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+ ```
+Filtered Disk Index
+  L   Beamwidth             QPS    Mean Latency    99.9 Latency        Mean IOs         CPU (s)       Recall@10
+==================================================================================================================
+ 10           4         1922.02         4062.19        12849.00           15.49           66.19           11.80
+ 20           4         4609.91         1618.68         3438.00           30.66          140.48           17.20
+ 30           4         3377.83         2250.22         4631.00           42.70          202.39           20.70
+ 40           4         2707.77         2817.21         4889.00           51.46          267.03           22.00
+ 50           4         2191.56         3509.43         5943.00           60.80          349.10           23.50
+100           4         1257.92         6113.45         7321.00          109.08          609.42           23.90
```
Comment (harsha-simhadri) [inline workflows/filtered_in_ssd.md:102]: Why is the recall this low?

File: src/index.cpp (hunk 3 above)
Comment (NeelamMahapatro) [reply src/index.cpp:564]: This assert statement is there just above the   line if (file_exists(labels_to_medoids)) condition.
Its seems duplicate

File: tests/build_memory_index.cpp (hunk 4 above)
Comment (NeelamMahapatro) [reply tests/build_memory_index.cpp:55]: The universal_label input can be a string and this conversion to int doesnot look correct.
Ideally, we should load the label map and find int label corresponding to the universal label. Should I do that ?

File: workflows/filtered_in_ssd.md (hunk 5 above)
Comment (NeelamMahapatro) [reply workflows/filtered_in_ssd.md:102]: Its similar to the in memory result - https://github.com/microsoft/DiskANN/blob/main/workflows/filtered_in_memory.md

Comment (harsha-simhadri) [reply workflows/filtered_in_ssd.md]: could you please change the params to offer produce recall in another PR?

**Pull request merged**